tool/src/main/java/org/datacommons/tool/Lint.java

```diff
@@ -54,56 +54,23 @@
 
   @Override
   public Integer call() throws IOException, InvalidProtocolBufferException {
-    List<File> mcfFiles = new ArrayList<>();
-    List<File> tmcfFiles = new ArrayList<>();
-    List<File> csvFiles = new ArrayList<>();
-    int nTsv = 0;
-    for (File file : files) {
-      String lowerPath = file.getPath().toLowerCase();
-      if (lowerPath.endsWith(".mcf")) {
-        mcfFiles.add(file);
-      } else if (lowerPath.endsWith(".tmcf")) {
-        tmcfFiles.add(file);
-      } else if (lowerPath.endsWith(".csv")) {
-        csvFiles.add(file);
-      } else if (lowerPath.endsWith(".tsv")) {
-        nTsv++;
-        csvFiles.add(file);
-      } else {
-        throw new CommandLine.ParameterException(
-            spec.commandLine(), "Found an unsupported file type: " + file.getPath());
-      }
-    }
-    logger.info(
-        "Input includes {} MCF file(s), {} TMCF file(s), {} CSV file(s)",
-        mcfFiles.size(),
-        tmcfFiles.size(),
-        csvFiles.size());
-    // Various checks
-    if (nTsv > 0 && nTsv != csvFiles.size()) {
-      throw new CommandLine.ParameterException(
-          spec.commandLine(), "Please do not mix .tsv and .csv files");
-    }
+
+    FileGroup fg = TmcfCsvReader.read(files, spec, logger);
```

Comment:
The class TmcfCsvReader gives the impression that it is actually reading the files, but it is just enumerating them. Moreover, this exists to just build a FileGroup, so how about making it a static function inside FileGroup?  

FileGroup.Build(...)

Reply:
Yea, that's better